src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/ExpressionConditionVariableResolverPass.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class ExpressionConditionVariableResolverPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    public function process(ContainerBuilder $container): void
+    {
+        if (false === $container->hasDefinition('routing.router')) {
+            return;
+        }
+
+        $routerDefinition = $container->getDefinition('routing.router');
+
+        foreach ($this->findAndSortTaggedServices('routing.expression_condition_variable', $container) as $reference) {
+            $serviceId = (string) $reference;
+            $serviceDefinition = $container->getDefinition($serviceId);
+            $tag = $serviceDefinition->getTag('routing.expression_condition_variable');
+            $name = $tag[0]['name'] ?? null;
```

Comment:
what does a `null` name mean?

Reply:
it means the tag is misconfigured

I don't know whether is possible to make sure a name is always valid at this point. I'm inclined to throw an exception here.

Reply:
this file was removed in eafc9c5749ac2063e5d9c3689a19202b12cbc6d1, since it injects a service locator (see `src/Symfony/Bundle/FrameworkBundle/Resources/config/routing.php`)